crates/biome_js_analyze/src/lint/nursery/no_label_without_control.rs

```diff
@@ -0,0 +1,290 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{
+    AnyJsxAttribute, AnyJsxAttributeName, AnyJsxAttributeValue, AnyJsxTag, JsxAttribute,
+    JsxAttributeList, JsxName, JsxReferenceIdentifier, JsxText,
+};
+use biome_rowan::AstNode;
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Enforce that a label element or component has a text label and an associated control.
+    ///
+    /// There are two supported ways to associate a label with a control:
+    /// - Wrapping a control in a label element.
+    /// - Adding a `for` attribute (or `htmlFor` in React) to a label and assigning it a DOM ID string that indicates an input on the page.
+    ///
+    /// This rule checks that any `label` element (or an indicated custom component that will output a `label` element) meets one of this conditions:
+    /// - Wraps an `input` element (or an indicated custom component that will output an `input` element)
+    /// - Has a `for` or `htmlFor` attribute and that the `label` element/component has accessible text content.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label for="js_id" />;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label for="js_id"><input /></label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label htmlFor="js_id" />;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label htmlFor="js_id"><input /></label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label>A label</label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div><label /><input /></div>;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
```

Comment:
For valid cases, you don't need multiple code blocks. One is enough.

Reply:
done!